### DISCUSSION TITLE
How do I get the row (id?) of a component in the skin edditor?

### DISCUSSION THREAD

**Discussion by deok420** [ppy/osu#20999] [Q&A]
Reference image: https://prnt.sc/iVYDQmBITkzj
I want to make it so component row 1 goes to component row 2's possition, and if I press a certain key that row to go to row 3's possition, but I have no idea how to access each individual row. At first I thought it was the ChildID of the component, but that wasn't the case. When in the Draw Visualizer, I could see that the ChildID corresponded to the number of the row the component was sitting on(for ex. ChildID:1 = row:1, ChildID:2 = row:2), but when I switched to Song Select components, the ChildID were 4 and 17, completely random. But either way, if they were actually correct, I still wouldn't be able to use them, because for some reason the int ChildId is not detected for components.
Sorry for this being too long, but I've been working on this for about 3 hours now and have gotten 0 progress.

**Comment by peppy** [comment]
I cannot understand this. Maybe share what you're attempting to do in the form of code?

**Reply by deok420** [reply]
All I want to know is if there is any way of moving components (on the component sidebar) from one row to another. I want to use this to make something like a layer system, because some legacy skins look off. I've been trying to get something out of the Draw Visualizer for 2 days now, but to no avail.
There is no code ready for this, I'm still searching. That's why I made this discussion. If you still can't understand what I'm trying to say, I'd be more than happy to make a concept type of video and send it. That way it'd be way easier to understand.
Sorry if this is a rookie question, I'm just really not used to working on other people's projects.

**Reply by peppy** [reply]
What is "off" in legacy skins that you're trying to fix? Please give full details.

**Reply by deok420** [reply]
An example I can give is Whitecat's skin (2.0), where the legacy song progress overlaps the little box where it's supposed to be under (Which comes from the Legacy health display). It's a really small thing, but I would still like to fix it.
![image](https://user-images.githubusercontent.com/90860478/198952204-7a7d189b-790c-4739-812f-85f404cc1cb3.png)

**Reply by peppy** [reply]
I'm not sure this can be amicably fixed. I'd recommend opening a fresh issue and attaching the skin so we can give further advice.

**Reply by deok420** [reply]
Sorry for responding so slow, I had some work to do yesterday. I submitted an issue.
